src/ui/CryptoUi.java

```diff
@@ -1,7 +1,58 @@
 package ui;
 
+import domain.CaesarsStrategy;
+import domain.CryptoContext;
+import domain.CryptoStrategy;
+import domain.ReflectionStrategy;
+
+import javax.swing.*;
+
 public class CryptoUi {
+    public void CryptoContext(){
+        CryptoContext cryptoContext;
+    }
+
     public void showUi(){
+        chooseSecretCode();
+    }
+
+    public void chooseSecretCode(){
+        String code = JOptionPane.showInputDialog("Enter your code to encode or decode:");
+        if (code.isEmpty()){
+            JOptionPane.showMessageDialog(null, "The code cannot be empty!");
+            return;
+        }
+        chooseOperation(code);
+    }
+
+    public void chooseOperation(String text){
```

Comment:
I think there was a method naming mismatch - chooseSecretCode meant choosing f.e. Caesars or Reflection strategy and chooseOperation was for encoding/decoding.